src/plone/restapi/serializer/atfields.py

```diff
@@ -32,9 +33,9 @@ def __init__(self, field, context, request):
         self.field = field
 
     def __call__(self):
-        accessor = self.field.getEditAccessor(self.context)
-        if accessor is None:
-            accessor = self.field.getAccessor(self.context)
+        # accessor = self.field.getEditAccessor(self.context)
```

Comment:
When using getEditAccessor, Archetypes internally calls to getRawText which returns the exact text saved on the field.

When calling getAccessor, the plone.outputfilters machinery gets hooked and runs the proper transform to convert resolveUID links into proper links.